staging/src/k8s.io/apiserver/pkg/server/openapi/openapi_aggregator.go

```diff
@@ -0,0 +1,462 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapi
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-openapi/spec"
+
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/apiserver/pkg/util/trie"
+)
+
+const (
+	DEFINITION_PREFIX = "#/definitions/"
+)
+
+var cloner = conversion.NewCloner()
+
+// Run a walkRefCallback method on all references of an OpenAPI spec
+type walkAllRefs struct {
+	walkRefCallback func(ref spec.Ref) spec.Ref
+	root            *spec.Swagger
+}
+
+func newWalkAllRef(walkRef func(ref spec.Ref) spec.Ref, sp *spec.Swagger) *walkAllRefs {
```

Comment:
It returns a reference that make it easier to call methods on (otherwise I need to store created object into a variable and call start upon it. I don't have a strong feeling to keep it around but it doesn't hurt to follow the model of constructor.